src/Domain/Tests/DomainCollectionTest.php

```diff
@@ -124,9 +127,11 @@ public function testLazySlice(): void
         self::assertSame([1 => 2], iterator_to_array(($collection = static::createLazyCollection([1, 2, 3, null, 5], $visited))->slice(1, 1)));
         self::assertSame([1, 2, 3], $visited);
 
-        $this->assertUnrewindableGenerator();
```

Comment:
if we can add a new test to cover this scenario (unrewindable) would be nice, but i can have a look at it later as well.